lib/fog/aws/requests/rds/modify_db_instance.rb

```diff
@@ -61,6 +61,9 @@ def modify_db_instance(db_name, apply_immediately, options={})
               #else
               #  modified_server = server["PendingModifiedValues"].merge!(options) # it appends
               #end
+              if options["NewDBInstanceIdentifier"]
+                options["DBInstanceIdentifier"] = options.delete("NewDBInstanceIdentifier")
```

Comment:
this should go in `PendingModifiedValues`, so it won't take effect until a reload.

see https://github.com/fog/fog-aws/blob/master/lib/fog/aws/requests/rds/describe_db_instances.rb#L58